packages/pymedphys_pinnacle/src/pymedphys_pinnacle/export/rtplan.py

```diff
@@ -0,0 +1,548 @@
+    datetimesplit = plan_info["ObjectVersion"]["WriteTimeStamp"].split()
+
+    # Read more accurate date from trial file if it is available
+    trial_info = plan.trial_info
+    if trial_info:
+        datetimesplit = trial_info['ObjectVersion']['WriteTimeStamp'].split()
+
+    ds.StudyDate = datetimesplit[0].replace('-', '')
+    ds.StudyTime = datetimesplit[1].replace(':', '')
+    ds.AccessionNumber = ''
+    ds.Modality = RTPLANModality
+    ds.Manufacturer = Manufacturer
+    ds.OperatorsName = ""
+    ds.ManufacturersModelName = plan_info["ToolType"]
+    ds.SoftwareVersions = [ plan_info["PinnacleVersionDescription"] ]
+    ds.PhysiciansOfRecord = patient_info["RadiationOncologist"]
+    ds.PatientName = patient_info["FullName"]
+    ds.PatientBirthDate = patient_info["DOB"]
+    ds.PatientID = patient_info["MedicalRecordNumber"]
+    ds.PatientSex = patient_info["Gender"][0]
+
+    ds.StudyInstanceUID = image_info["StudyInstanceUID"]
+    ds.SeriesInstanceUID = planInstanceUID
+    ds.StudyID = plan.primary_image.image["StudyID"]
+
+    ds.FrameOfReferenceUID = image_info["FrameUID"]
+    ds.PositionReferenceIndicator = ""
+
+    ds.RTPlanLabel = plan.plan_info['PlanName'] + '.0'
+    ds.RTPlanName = plan.plan_info['PlanName']
+    ds.RTPlanDescription = plan.pinnacle.patient_info['Comment']
+    ds.RTPlanDate = ds.StudyDate
+    ds.RTPlanTime = ds.StudyTime
+
+    # ds.PlanIntent = "" #Not sure where to get this informationd, will likely
+    # be 'CURATIVE' or 'PALIATIVE'
+    ds.RTPlanGeometry = 'PATIENT'
+    # ds.DoseReferenceSequence = Sequence() #figure out what goes in DoseReferenceSequence... Should be like a target volume and reference point I think...
+    # ds.ToleranceTableSequence = Sequence() #figure out where to get this
+    # information
+    ds.FractionGroupSequence = Sequence()
+    ds.BeamSequence = Sequence()
+    ds.PatientSetupSequence = Sequence()  # need one per beam
+    ds.ReferencedStructureSetSequence = Sequence()
+    ReferencedStructureSet1 = Dataset()
+    ds.ReferencedStructureSetSequence.append(ReferencedStructureSet1)
+    ds.ReferencedStructureSetSequence[0].ReferencedSOPClassUID = RTStructSOPClassUID
+    ds.ReferencedStructureSetSequence[0].ReferencedSOPInstanceUID = plan.struct_inst_uid
+    ds.ApprovalStatus = 'UNAPPROVED'  # find out where to get this information
+
+    ds.FractionGroupSequence.append(Dataset())
+    ds.FractionGroupSequence[0].ReferencedBeamSequence = Sequence()
+
+    metersetweight = ['0']
+
+    num_fractions = 0
+    beam_count = 0
+    beam_list = trial_info['BeamList'] if trial_info['BeamList'] else []
+    if len(beam_list) == 0:
+        plan.logger.warning("No Beams found in Trial. Unable to generate RTPLAN.")
+        return None
+    for beam in beam_list:
+
+        beam_count = beam_count + 1
+
+        plan.logger.info("Exporting Plan for beam: " + beam['Name'])
+
+        ds.PatientSetupSequence.append(Dataset())
```

Comment:
Instead of appending this at the start and then having to reference it with `ds.PatientSetupSequence[beam_count - 1]` call it `patient_setup` and append it once it is built.